common/arg.cpp

```diff
@@ -220,12 +226,25 @@ struct curl_slist_ptr {
 #define CURL_MAX_RETRY 3
 #define CURL_RETRY_DELAY_SECONDS 2
 
-static bool curl_perform_with_retry(const std::string & url, CURL * curl, int max_attempts, int retry_delay_seconds, const char * method_name) {
+static bool curl_perform_with_retry(const std::string & url,
```

Comment:
@ngxson I suspect you could have written a lot of this. This could be a follow on PR. But is there an argument to be made that we should check things like "Last-Modified" header before each retry? For the rare case that the model got updated in between requests.